transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/FileTransformations.scala

```diff
@@ -98,9 +98,74 @@ object FileTransformations {
 
   /** Compute the data modality of a raw file. */
   private def computeDataModality(rawFile: Msg): String =
-    rawFile match {
-      // FIXME: Actually compute something here once we know the formula.
-      case _ => "I'm a modality!"
+    rawFile.tryRead[String]("assay_term_name") match {
+      case None                            => ""
```

Comment:
If there's ever a case where we can't determine the correct modality, we should output `None` instead of a placeholder value. You'll need to update the schemas of the file tables to remove the `required` modifier on each column.

Reply:
This can then become slightly less verbose:
```scala
rawFile.tryRead[String]("assay_term_name").collect {
  case "3' RACE" => "Transcriptomic"
  .. etc ...
}
```